lib/travis/services/find_build.rb

```diff
@@ -46,8 +46,10 @@ def load_result
 
         def preload(build)
           ActiveRecord::Associations::Preloader.new(build, [:matrix, :commit, :request]).run
-          ActiveRecord::Associations::Preloader.new(build.matrix, :log, select: [:id, :job_id, :updated_at]).run
           ActiveRecord::Associations::Preloader.new(build.matrix, :annotations).run
+          unless Travis.config.logs_api.enabled?
```

Comment:
These suggestions about using a statement modifier might make the line really long, but I personally prefer that to a dangling end.